arwen/host/execution.go

```diff
@@ -411,6 +421,21 @@ func (host *vmHost) callSCMethodIndirect() error {
 	return nil
 }
 
+func (host *vmHost) callBuiltinFunction(input *vmcommon.ContractCallInput) error {
+	_, _, metering, output, runtime, _ := host.GetContexts()
+
+	valueToSend, gasConsumed, err := host.blockChainHook.ProcessBuiltInFunction(input)
+	if err != nil {
+		metering.UseGas(gasConsumed)
+		return err
+	}
+
+	scAccount, _ := output.GetOutputAccount(runtime.GetSCAddress())
+	scAccount.BalanceDelta.Add(scAccount.BalanceDelta, valueToSend)
```

Comment:
Just a question - this is always addition, correct? Or perhaps `valueToSend` could also be negative?

Reply:
it does not matter. add can do minus as well. built-in functions is has a tons of checks.